controllers/apps/transformer_rbac.go

```diff
@@ -57,8 +60,25 @@ func (c *RBACTransformer) Transform(ctx graph.TransformContext, dag *graph.DAG)
 	if err != nil {
 		return err
 	}
+	componentSpecs := cluster.Spec.ComponentSpecs
+	if componentSpecs == nil {
```

Comment:
why componentSpecs is nil means that there will be no new component built from cluster template?

Reply:
if componentSpecs is nil, and cluster template has a corresponding part, new component will be generated from that

Reply:
sorry, the above comments missed a "not", there is scenario that componentSpecs is not nil, but it will still be possible to build a new component from cluster template.